src/Components/WebAssembly/Build/src/targets/Blazor.MonoRuntime.targets

```diff
@@ -318,12 +320,43 @@
     </ItemGroup>
   </Target>
 
+  <Target Name="_GenerateBlazorBootJsonInputHash">
+    <ItemGroup>
+      <_BlazorBootJsonHashInput Include="@(IntermediateAssembly)" />
+      <_BlazorBootJsonHashInput Include="@(_BlazorOutputWithTargetPath)" />
+      <_BlazorBootJsonHashInput Include="@(_BlazorConfigFile)" />
+    </ItemGroup>
+
+    <Hash ItemsToHash="@(_BlazorBootJsonHashInput)">
+      <Output TaskParameter="HashResult" PropertyName="_BlazorBootJsonInputHash" />
+    </Hash>
+
+    <PropertyGroup>
+      <_BlazorBootJsonInputHashFile>$(_BlazorIntermediateOutputPath)boot.json.input</_BlazorBootJsonInputHashFile>
+    </PropertyGroup>
+
+    <WriteLinesToFile
+      Lines="$(_BlazorBootJsonInputHash)"
+      File="$(_BlazorBootJsonInputHashFile)"
+      Overwrite="True"
+      WriteOnlyWhenDifferent="True" />
+
+  </Target>
+
   <UsingTask TaskName="GenerateBlazorBootJson" AssemblyFile="$(_BlazorTasksPath)" />
 
   <Target
     Name="_GenerateBlazorBootJson"
-    Inputs="$(MSBuildAllProjects);@(_BlazorOutputWithTargetPath)"
+    DependsOnTargets="_GenerateBlazorBootJsonInputHash"
+    Inputs="$(MSBuildAllProjects);@(_BlazorOutputWithTargetPath);$(_BlazorBootJsonInputHashFile)"
```

Comment:
I would have thought that `@(_BlazorConfigFile)` should be included in the inputs here, not just `$(_BlazorBootJsonInputHashFile)`, but please let me know what I'm misunderstanding!

Reply:
The hash input should cover the list of config files, which is what the boot.json cares about. The config files are always read uncached by the browser - unlike `_BlazorOutputWithTargetPath`, we only need to regenerate it if the list of config files changes, not their contents.

Reply:
I see. So we can do this for these files precisely because we're *not* embedding a content hash for them in the boot manifest. Makes sense.